### PR TITLE
Update StableMIR doc to reflect current status

### PR DESCRIPTION
**Pull request by celinval**
Update stable-mir documentation, since we no longer use git subtree, and we have 2 different crates.

### CODE REVIEW

File: compiler/stable_mir/README.md [hunk 1]
```diff
@@ -1,93 +1,25 @@
...
-
-```
-git subtree pull --prefix=compiler/rustc_smir https://github.com/rust-lang/project-stable-mir smir
-```
-
-Note: only ever sync to rustc from the project-stable-mir's `smir` branch. Do not sync with your own forks.
-
-Then open a PR against rustc just like a regular PR.
+Our goal is to start publishing `stable_mir` into crates.io.
+Until then, users will use this as any other rustc crate, via extern crate.
```
Comment (compiler-errors) [inline compiler/stable_mir/README.md:4]: Perhaps mention that users need to install the `rustc-dev` component.

File: compiler/stable_mir/README.md [hunk 2]
```diff
@@ -1,93 +1,25 @@
...
 
 ## Stable MIR Design
 
 The stable-mir will follow a similar approach to proc-macro2. It’s
```
Comment (compiler-errors) [inline compiler/stable_mir/README.md:8]:
```suggestion
The stable-mir will follow a similar approach to proc-macro2. Its
```

File: compiler/stable_mir/README.md [hunk 3]
```diff
@@ -1,93 +1,25 @@
...
-implementation will eventually be broken down into two main crates:
+implementation is done using two main crates:
```
Comment (compiler-errors) [inline compiler/stable_mir/README.md:9]:
```suggestion
implementation is split between two main crates:
```

File: compiler/stable_mir/README.md [hunk 4]
```diff
@@ -1,93 +1,25 @@
...
 
 - `stable_mir`: Public crate, to be published on crates.io, which will contain
```
Comment (compiler-errors) [inline compiler/stable_mir/README.md:11]:
```
- `stable_mir`: Public crate, to be published on crates.io, whics contains
the stable data structures and APIs to access them. It calls to `rustc_smir`.
- `rustc_smir`: This crate serves as `stable_mir`'s interface to the compiler.
It is responsible for gathering all the information requested, and converting
them to their stable data structures.
```

Comment (compiler-errors) [reply compiler/stable_mir/README.md:11]: Github didn't allow me to rewrite this as an inline suggestion, but this section seems wrong. AFAICT, the `rustc_smir` crate is what does the conversion; hence it having all the `.stable` calls.

**Review by compiler-errors — CHANGES_REQUESTED** [review]
Some nits. Let me know if I misunderstood something, particularly in the "Stable MIR design" section.

**Comment by compiler-errors** [comment]
Anyways r=me after tweaking unless you specifically want a review from scott

File: compiler/stable_mir/README.md (hunk 4 above)
Comment (celinval) [reply compiler/stable_mir/README.md:11]: Indeed. Not for long though, but I'll update this to match the current state. Thanks

Comment (celinval) [reply compiler/stable_mir/README.md:15]: I documented that we are in a transition period. Let me know if that makes sense.

**Comment by celinval** [comment]
@compiler-errors do you mind taking a quick look at the updated version? Thanks

**Review by compiler-errors — APPROVED** [review]

**Pull request merged**